internal/hcsoci/drivers.go

```diff
@@ -0,0 +1,109 @@
+// +build windows
+
+package hcsoci
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/oci"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/pkg/errors"
+)
+
+// getAssignedDeviceUtilityTool gets the path to device-util on the server host
+func getAssignedDeviceUtilityTool(coi *createOptionsInternal) (string, error) {
+	tools, ok := coi.Spec.Annotations[oci.AnnotationAssignedDeviceUtilityTool]
+	if !ok || tools == "" {
+		return "", fmt.Errorf("no driver tools were specified for %s", coi.actualID)
+	}
+	if _, err := os.Stat(tools); err != nil {
+		return "", errors.Wrapf(err, "failed to find device installation tools at %s", tools)
+	}
+	return tools, nil
+}
+
+// getAssignedDeviceKernelDrivers gets any device drivers specified on the spec.
+// Drivers are optional, therefore do not return an error if none are on the spec.
+func getAssignedDeviceKernelDrivers(coi *createOptionsInternal) ([]string, error) {
+	csDrivers, ok := coi.Spec.Annotations[oci.AnnotationAssignedDeviceKernelDrivers]
+	if !ok || csDrivers == "" {
+		return nil, nil
+	}
+	drivers := strings.Split(csDrivers, ",")
+	for _, driver := range drivers {
+		if _, err := os.Stat(driver); err != nil {
```

Comment:
What should these paths be pointing to (installers, .ini files, .sys files, etc.)? That would be good to make clear here.

Reply:
Sure I'll add a comment. They should point to directories that contain .inf, .sys, .cer, etc files that are needed during standard pnputil driver installation.

Reply:
Done.